api/build/enqueue.go

```diff
@@ -16,28 +16,36 @@ import (
 
 // Enqueue is a helper function that pushes a queue item (build, repo, user) to the queue.
 func Enqueue(ctx context.Context, queue queue.Service, db database.Interface, item *models.Item, route string) {
-	logrus.Infof("Converting queue item to json for build %d for %s", item.Build.GetNumber(), item.Build.GetRepo().GetFullName())
+	logger := logrus.WithFields(logrus.Fields{
+		"build":    item.Build.GetNumber(),
+		"build_id": item.Build.GetID(),
+		"org":      item.Build.GetRepo().GetOrg(),
+		"repo":     item.Build.GetRepo().GetName(),
+		"repo_id":  item.Build.GetRepo().GetID(),
+	})
+
+	logger.Debug("converting queue item to json")
 
 	byteItem, err := json.Marshal(item)
 	if err != nil {
-		logrus.Errorf("Failed to convert item to json for build %d for %s: %v", item.Build.GetNumber(), item.Build.GetRepo().GetFullName(), err)
+		logger.Errorf("failed to convert item to json: %v", err)
 
 		// error out the build
 		CleanBuild(ctx, db, item.Build, nil, nil, err)
 
 		return
 	}
 
-	logrus.Infof("Pushing item for build %d for %s to queue route %s", item.Build.GetNumber(), item.Build.GetRepo().GetFullName(), route)
+	logger.Debugf("pushing item for build to queue route %#q", route)
```

Comment:
whats the reasoning for the raw string literal`%#q` ? just curious

Reply:
i was starting to play with introducing this pattern. it's a bit nicer than straight up `%s`. you can avoid things like `queue route route` (if the route is named `route`) or `queue route ` (if the var is empty). it would be `` queue route `route` `` and `` queue route `​` `` respectively. it also, of course, safely escapes the string. i would like to use it over `%s`. 

i reverted my changes for that (but missed that spot), to maybe introduce later. i will fix this instance up.

Reply:
i reverted this